File: src/models/Item.ts
import { Section } from "./Section";

class Item extends Section{
    checked: boolean;
    
    constructor(name: string, parent: Section){
        super();
        this.name = name;
        this.parent = parent;
        this.checked = false;
    }
    
    sortItemList(unorderedList: Item[]): [Item[], Item[]] {
        // console.log(`Item iput: ${this.name}`)
        // console.log(unorderedList)
        let ordered: Item[] = [];
        if (unorderedList.length != 0 
            && unorderedList.find((item) => item.name === this.name)) {
                ordered.push(this)
                unorderedList = unorderedList.filter((item) => item.name !== this.name)
            }
        // console.log(`Item return: ${this.name}`)
        // console.log([ordered, unorderedList])
        return [ordered, unorderedList]
    }
    
}

export {Item}

File: app/src/main/java/com/example/walmartpathfinder/models/Item.kt
package com.example.walmartpathfinder.models

class Item(name: String, parent: Section) : Section(name, parent) {
    /* Check list to see if item is present in unordered list and add to ordered list */
    override fun sortItemList(unorderedList: MutableList<Item>): Pair<MutableList<Item>, MutableList<Item>> {
        var ordered = mutableListOf<Item>()

        if (this in unorderedList) {
            ordered.add(this)
            unorderedList.remove(this)
        }

        return Pair(ordered, unorderedList)
    }
}